test/test.js

```diff
@@ -1,12 +1,68 @@
-var request = require('supertest');
+var supertest = require('supertest');
+var db = require('./db');
+
+var request = function (path) {
+    return supertest('http://localhost:6969')
+    .get(path)
+    .expect(200);
+}
+
+describe.skipIfImpl = function(impls, title, fn) {
+  var currentImpl = process.env["EBOSHI_API_SHOOTOUT_CURRENT_IMPL"];
+  if(impls.indexOf(currentImpl) !== -1) {
+      return this.skip(title, fn);
+  } else {
+      return this(title, fn);
+  };
+};
 
 describe('/api/test', function() {
+    it('returns "Hello world"', function (done) {
+        request('/api/test').expect('Hello world', done);
+    });
+});
 
-    it('should return Hello world', function (done) {
-        request('http://localhost:6969')
-        .get('/api/test')
-        .expect(200)
-        .expect('Hello world', done);
+describe.skipIfImpl(['./elixir_phoenix', './node_express', './ruby_sinatra'], '/api/clients', function() {
+    before(function(done) { db.bootstrap(done); });
+
+    before(function(done) {
+        db.query(
+            "INSERT INTO clients SET " +
+            "  name='Bot and Rose Design', " +
+            "  address='625 NW Everett St', " +
+            "  city='Portland', " +
+            "  state='OR', " +
+            "  zip='97209', " +
+            "  country='USA', " +
+            "  email='info@botandrose.com', " +
+            "  contact='Michael Gubitosa', " +
+            "  phone='(503) 662-2712', " +
+            "  created_at='2006-06-25T14:08:31Z', " +
+            "  updated_at='2015-08-29T09:58:23Z';",
```

Comment:
Well, my original plan was to write the tests in shell so that the test suite didn't have any dependencies; itd just run on anything unixy out of the box. Shell turned out to be too painful, so I switched to node 0.10+, since most machines seem to have at least that installed as a baseline, and it speaks json natively. Gonna try to avoid introducing more complexity into the test set up if I can, but I may end up giving up that ideal as well... we'll see how it goes.